### PR TITLE
Update *[MO] Deploy to staging* with AppCenter

### PR DESCRIPTION
**Pull request by felixmeziere**

### CODE REVIEW

**Comment by bam-bot-tech** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  **<a href="https://github.com/bamlab/dev-standards/blob/mo-deploy-to-staging-app-center/react-native/setup/setup-and-deploy-new-project-to-staging-with-hockeyapp.mo.md">setup-and-deploy-new-project-to-staging-with-hockeyapp.mo.md</a>**: Seems you do not need a *Troubleshoot* part
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **<a href="https://github.com/bamlab/dev-standards/blob/mo-deploy-to-staging-app-center/react-native/setup/setup-and-deploy-new-project-to-staging.mo.md">setup-and-deploy-new-project-to-staging.mo.md</a>**: Seems you do not need a *Troubleshoot* part
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by louiszawadzki** [comment]
@felixmeziere our current flow still uses Hockey App, I suggest we keep the MO as it is until mid-2018 as 

> We’re working on the specific timing, but currently we expect the option to fully migrate off of HockeyApp to App Center to be available in mid-2018.

(https://hockeyapp.net/appcenter/faq/)

Meanwhile you can keep the PR open so we can build on it as we integrate App Center.

**Review by tychota — COMMENTED** [review]
Your file is named:

"react-native/setup/setup-and-deploy-new-project-to-staging.md"

where it should be:

react-native/setup/setup-and-deploy-new-project-to-staging.**mo**.md

I know that you only have modified it, but now that we have a poka yoke to catch this, I would like you to fix it.

**Comment by tychota** [comment]
ping @felixmeziere !

**Comment by felixmeziere** [comment]
@louiszawadzki I think we should do the move before they migrate completely, this is the point of the standard. We need to be ready for when that happens.
We are already using AppCenter in the UK and using it is seamless.
Have you noticed any feature that you used in HockeyApp but missing in AppCenter?

Also, the idea that we mentioned with Yann was to rename this standard to "[MO] Deploy to staging with AppCenter" and have the two in parallel for a little while, then delete the current one.

Are you ok with this?

@tychota I'll address the change ASAP

**Comment by tychota** [comment]
@felixmeziere I would like instead having a "[MO] Deploy to staging" which points to this standard but move the old standard to "[MO] Deploy to staging with HockeyApp".

For the depreciation you can use hints. The syntax is something like:

for `deploy-to-staging-with-hockey.mo.md`:

```
# [MO] Deploy to staging with HockeyApp

{% hint style='warning' } **DEPRECIATION**

This standard has been deprecated by [@felixmeziere](link/to/your/github/profile) on January 24 in favour of [[MO] Deploy to staging](link to the new standard).

**Reason**: hockeyapp is mean to be deprecated by appcenter is the new hockey, 100% compatible with hockey.

{% endhint %}

/// rest of the standard
```

for `deploy-to-staging.mo.md`:

```
# [MO] Deploy to staging

{% hint style='warning' } **NOTE**

This standard replaces [[MO] Deploy to staging with hockeyapp](link to the old standard), deprecated by [@felixmeziere](link/to/your/github/profile) on January 24.

**Reason**: hockeyapp is mean to be deprecated by appcenter is the new hockey, 100% compatible with hockey.

{% endhint %}

/// rest of the standard
```

That produce nice box like https://bamtech.gitbook.io/dev-standards/react-native/redux/mo-pass-props-to-a-container#steps:

screen shot 2018-01-24 at 19 44 08

**Comment by tychota** [comment]
(Wrong button sorry)

**Comment by felixmeziere** [comment]
Thanks! I'll do it.
Have you solved the problem of the link for the PO?

File: react-native/setup/setup-and-deploy-new-project-to-staging-with-hockeyapp.mo.md [hunk 1]
```diff
@@ -1,4 +1,12 @@
-# [MO] Setup & Deploy New Project to Staging
+# [MO] Setup & Deploy New Project to Staging with HockeyApp
+
+{% hint style='warning' } **DEPRECATION**
+
+This standard has been deprecated by [@felixmeziere](https://github.com/felixmeziere) on January 28 in favour of [[MO] Deploy to staging](./setup-and-deploy-new-project-to-staging.mo.md).
+
+**Reason**: HockeyApp is meant to be deprecated by AppCenter. AppCenter is the new HockeyApp and there is [no loss of functionality](http://blog.m33.network/2017/09/react-native-devops-2-0-overview-of-mobile-center-next-generation-of-hockeyapp/).
```
Comment (tychota) [inline react-native/setup/setup-and-deploy-new-project-to-staging-with-hockeyapp.mo.md:7]: Is the link accessible to public (not M33) ?

Comment (tychota) [reply react-native/setup/setup-and-deploy-new-project-to-staging-with-hockeyapp.mo.md:7]: And you can put @yleflour as owner of the deprecated standard.

File: react-native/setup/setup-and-deploy-new-project-to-staging.mo.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
+# [MO] Setup & Deploy New Project to Staging
+
+{% hint style='warning' } **NOTE**
```
Comment (tychota) [inline react-native/setup/setup-and-deploy-new-project-to-staging.mo.md:3]: You can put yourself as owner of the new standard (see other standards to see the format).

File: react-native/setup/setup-and-deploy-new-project-to-staging.mo.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+{% endhint %}
+
+## Prerequisites
+- [ ] Have you entire environment setup
+- [ ] Generating a new SSH key and adding it to the ssh-agent: (https://help.github.com/articles/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent/)
+
+## Steps
+
+> :warning: Remember to commit after each step
```
Comment (tychota) [inline react-native/setup/setup-and-deploy-new-project-to-staging.mo.md]: The :warning: can be changed to 

{% hint style='warning %}

File: react-native/setup/setup-and-deploy-new-project-to-staging.mo.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+### 1. Setup your React Native App
+
+```
```
Comment (tychota) [inline react-native/setup/setup-and-deploy-new-project-to-staging.mo.md]: bash

File: react-native/setup/setup-and-deploy-new-project-to-staging.mo.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+react-native init <projectName>
+cd <projectName>
+# First Commit
+```
+
+### 2. Setup Fastlane
+
+```
```
Comment (tychota) [inline react-native/setup/setup-and-deploy-new-project-to-staging.mo.md]: bash

File: react-native/setup/setup-and-deploy-new-project-to-staging.mo.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+***Answers***
+- Please confirm the project name: `<Press Enter>`
+- Commit keystore files?: `Y`
+- Overwrite <fileName>: `<Press Enter>`
+
+### 3. Setup Staging Env
+
+```
```
Comment (tychota) [inline react-native/setup/setup-and-deploy-new-project-to-staging.mo.md]: Bash

File: react-native/setup/setup-and-deploy-new-project-to-staging.mo.md [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+- Your keystore password: `<Press Enter>`
+- A valid App Center API token: `**redacted**`
+- A valid App Center Username: `**redacted**`
+- (After some npm installation...) Should fastlane modify the Gemfile at path 'xxx' for you? (y/n): `y`
+
+*Note:* The AppCenter username is at the bottom-left of the AppCenter interface for a person or is the name of the organization for an organization.
+
+
+### 4. Deployment setup
+```
```
Comment (tychota) [inline react-native/setup/setup-and-deploy-new-project-to-staging.mo.md]: bash and empty line between title and code

File: react-native/setup/setup-and-deploy-new-project-to-staging.mo.md [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+bundle exec fastlane ios setup --env=staging
+# Fourth Commit
+```
+
+### 5. Deploy Staging
+```
```
Comment (tychota) [inline react-native/setup/setup-and-deploy-new-project-to-staging.mo.md]: bash and empty line between title and code

File: react-native/setup/setup-and-deploy-new-project-to-staging.mo.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+bundle exec fastlane android deploy --env=staging
+```
+
+***Answers***
+- OS: `iOS/Android` depending on which you are deploying
+- Platform: `React Native`
+- Do you want to create a New App?: `yes`
+
+### 6. Get the download link
+- For each app (Android + iOS)
```
Comment (tychota) [inline react-native/setup/setup-and-deploy-new-project-to-staging.mo.md]: Empty line

**Review by tychota — CHANGES_REQUESTED** [review]
It is very good @felixmeziere. Just some markdown nits and danger auto-checks.

image


You shall had owner to the standards (you can add yourself to the new and yann to the old).

And you hall double check that every standard are included in the readme and the summary file (even the deprecated one), else gitbook is not going to found them.

File: react-native/setup/setup-and-deploy-new-project-to-staging-with-hockeyapp.mo.md (hunk 1 above)
Comment (felixmeziere) [reply react-native/setup/setup-and-deploy-new-project-to-staging-with-hockeyapp.mo.md:7]: no the link isn't accessible to public.

**Pull request merged**